app/models/facilities_management/procurement.rb

```diff
@@ -393,7 +412,13 @@ def all_services_missing_benchmark_price?
     end
 
     def contract_value_needed?
-      all_services_unpriced_and_no_buyer_input? && !lot_number_selected_by_customer
+      (all_services_unpriced_and_no_buyer_input? || some_services_unpriced_and_no_buyer_input?) && !lot_number_selected_by_customer
+    end
+
+    def percentage_of_services_missing_framework_and_benchmark_price
+      uniq_building_services = procurement_building_services.all.uniq
+      unpriced_services = uniq_building_services.select { |pbs| CCS::FM::Rate.benchmark_rate_for(pbs.code, pbs.service_standard).nil? || CCS::FM::Rate.framework_rate_for(pbs.code, pbs.service_standard).nil? }
+      unpriced_services.size / uniq_building_services.size
```

Comment:
Dividing an integer by another integer will result in integer arithmetic - I'm assuming that's not what you want. If you want a float cast one of the operands to a float. E.g.
```suggestion
      unpriced_services.size / uniq_building_services.size.to_f
```
Also is it a fraction or a percentage you're after? Multiply by 100 if you want a percentage.